utils/src/main/java/software/amazon/awssdk/utils/BinaryUtils.java

```diff
@@ -117,6 +117,60 @@ public static ByteArrayInputStream toStream(ByteBuffer byteBuffer) {
         return new ByteArrayInputStream(copyBytesFrom(byteBuffer));
     }
 
+    /**
+     * Returns an immutable copy of the given {@code ByteBuffer}.
+     * <p>
+     * The new buffer's position will be set to the position of the given {@code ByteBuffer}, but the mark if defined will be
+     * ignored.
+     * <p>
+     * Note: this method intentionally converts direct buffers to non-direct, see
+     * <a href="https://github.com/aws/aws-sdk-java-v2/issues/3928">issue #3928</a>
```

Comment:
Nit: I don't know if we should bake this intention into this method, since it will likely be used elsewhere in the future. Can we just say that this is the current behavior, and leave it at that?

Reply:
should I add a note that this may change in future?

Reply:
done